### PR TITLE
Deprecate Configuration usages

### PR DESCRIPTION
**Pull request by FabienPapet**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Configuration in adapter is instanciated by hand, this PR deprecates it when possible
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | yes
| How to test?      | CI 🟢

### CODE REVIEW

**Review by nicosomb — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by FabienPapet** [comment]
tests: 
https://github.com/FabienPapet/ga.tests.ui.pr/actions/runs/4053291016

File: src/Core/Stock/StockManager.php [hunk 1]
```diff
@@ -389,8 +389,9 @@ private function prepareMovement($productId, $productAttributeId, $deltaQuantity
             $stockAvailable = $stockManager->getStockAvailableByProduct($product, $productAttributeId, $params['id_shop'] ?? null);
 
             if ($stockAvailable->id) {
+                $configuration = ServiceLocator::get('\\PrestaShop\\PrestaShop\\Core\\ConfigurationInterface');
```
Comment (jolelievre) [inline src/Core/Stock/StockManager.php:392]: This is related to the block below `if (!empty($params['id_stock_mvt_reason'])) {` it should probably be executed only in an else statement

File: src/PrestaShopBundle/Entity/StockMvt.php [hunk 2]
```diff
@@ -146,6 +146,8 @@ class StockMvt
 
     public function __construct()
     {
+        @trigger_error('This function is deprecated since 8.1 and will be dropped in 9.0.', E_USER_DEPRECATED);
```
Comment (jolelievre) [inline src/PrestaShopBundle/Entity/StockMvt.php:149]: UUurrrh what the fuck was that? 😅

**Review by jolelievre — COMMENTED** [review]
Just a small comment

**Pull request closed without merging**